packages/neos-ui-guest-frame/src/initializeGuestFrame.js

```diff
@@ -64,17 +64,33 @@ export default ({globalRegistry, store}) => function * initializeGuestFrame() {
     // Load legacy node data scripts from guest frame - remove with Neos 9.0
     const legacyNodeData = guestFrameWindow['@Neos.Neos.Ui:Nodes'] || {};
 
-    // Load all nodedata for nodes in the guest frame
+    // Load all nodedata for nodes in the guest frame and filter duplicates
     const {q} = yield backend.get();
     const nodeContextPathsInGuestFrame = findAllNodesInGuestFrame().map(node => node.getAttribute('data-__neos-node-contextpath'));
-    const fullyLoadedNodesFromContent = (yield q(nodeContextPathsInGuestFrame).get()).reduce((nodes, node) => {
+
+    // Filter nodes that are already present in the redux store and duplicates
+    const nodesByContextPath = store.getState().cr.nodes.byContextPath;
+    const nodesAlreadyPresentInStore = {};
+    const notFullyLoadedNodeContextPaths = [...new Set(nodeContextPathsInGuestFrame)].filter((contextPath) => {
+        const node = nodesByContextPath[contextPath];
+        const nodeIsLoaded = node !== undefined && node.isFullyLoaded;
+        if (nodeIsLoaded){
+            nodesAlreadyPresentInStore[contextPath] = node;
+            return false;
+        }
+        return true;
+    });
+
+    // Load remaining list of not fully loaded nodes from the backend if there are any
+    const fullyLoadedNodesFromContent = notFullyLoadedNodeContextPaths.length > 0 ? (yield q(notFullyLoadedNodeContextPaths).get()).reduce((nodes, node) => {
         nodes[node.contextPath] = node;
         return nodes;
-    }, {});
+    }, {}) : [];
 
     const nodes = Object.assign(
         {},
         legacyNodeData, // Merge legacy node data from the guest frame - remove with Neos 9.0
+        nodesAlreadyPresentInStore,
```

Comment:
as we merge we dont need to include them here imo ... they are already in the store